cmd/subctl/diagnose.go

```diff
@@ -211,41 +213,6 @@ func addDiagnoseFWConfigFlags(command *cobra.Command) {
 		"time to run in seconds while validating the firewall")
 	command.Flags().BoolVar(&diagnoseFirewallOptions.VerboseOutput, "verbose", false,
 		"produce verbose output while validating the firewall")
-	addDiagnosePodNamespaceFlag(command, &diagnoseFirewallOptions.PodNamespace)
```

Comment:
I didn't understand why this call was moved out of this function, it just seems to add code duplication.

Reply:
This function is called on all `firewall` sub-commands, but `inter-cluster` and `nat-discovery` now use the shared namespace support, so we can’t add a specific namespace flag there. As a result, the `addDiagnosePodNamespaceFlag` call must only be made on the other sub-commands.